### ISSUE TITLE
Function depends on project name

### ISSUE THREAD

**Issue by mnlipp** [eclipse/wildwebdeveloper#785]
I observe an extremely strange behavior in my workspace (which contains several projects). If the project name starts with [a-zA-I], no errors are shown for my ts-file. If the project name starts with [J-Z], it works.

I cannot reproduce this with an empty workspace and don't expect a solution. But I cannot find any documentation how to debug WWD. Can anybody provide a hint how to enable some debug logging?

**Comment by mickaelistria** [comment]
First step towards debugging is to identify steps to reproduce the issue. There are multiple approaches to logging/debugging, and it's hard to figure out which one to get started without steps to reproduce.

**Comment by mnlipp** [comment]
Agreed. But where should I get started with this strange behavior? All I can see is the dependency on the project name.

My idea was to enable some logging and search for differences when starting eclipse with different project names.

**Comment by mickaelistria** [comment]
Please share steps to reproduce when you have some and then we may be able to look at it and provide an answer.

**Comment by mnlipp** [comment]
Is there really no way to enable some logging? The project in question is OSS, so their is no problem in allowing you to reproduce this. But it is a complex project and without some information from WWD, I have no idea how to simplify this (detecting the dependency on the first letter of the project name already took hours and happened by chance).

What's the developer setup for WWD? Start the extension in a new instance from PDE and use the debugger, simple as that? Anything special to observe when doing this? Any hints for getting started?

**Comment by mickaelistria** [comment]
> What's the developer setup for WWD? Start the extension in a new instance from PDE and use the debugger, simple as that? Anything special to observe when doing this? Any hints for getting started?

Get the code, build it with Maven to get the dependencies included, and then you can start it as a regular plugin from your workspace.
But it's a client-server system, so just having debug/logging on client side may not be sufficient. You may need to also debug the language server. You can achieve it by tweaking https://github.com/eclipse/wildwebdeveloper/blob/master/org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/jsts/JSTSLanguageServer.java in your workspace to enable node debugging and connecting to it.
Debugging all that would be very complex and take much time, and I really think you should first invest in trying to simplify the example and identifying some steps to reproduce before debugging.

**Comment by mnlipp** [comment]
Well, okay. Thanks for the offer. (BTW: I'm using Linux (Fedora), sometimes this matters.)

 * Checkout https://github.com/mnlipp/wwd-problem
 * Run `npm install` in the root
 * Run `npm install` in aash-vue-components
 * Import the project in Eclipse (I've tested this with the latest WWD-distribution)
 * In the nested project `aash-vue-components` open `test.ts` --> no error shown
 * Close `test.ts` again
 * Rename project `aash-vue-Components` to `AAA`
 * Open `test.js` again --> errors are shown

I know that there are still a lot of node/typescript whatever related files lying around there (which only make sense when you look at the [full project](https://github.com/mnlipp/jgrapes-webconsole), but there they are needed). My problem is that I have no idea where to go (i.e. simplify) from here. I could start to arbitrarily delete or change configuration files. But apart from this being groping in the dark, every change in these files will break my project -- they are there for a reason.

**Comment by mickaelistria** [comment]
On step 5, I see errors on the file before a rename. Can you please try latest snapshots of Wild Web Developer (upgrading to https://download.eclipse.org/wildwebdeveloper/snaspshots )? Maybe it's an issue that was fixed since last release.

**Comment by mnlipp** [comment]
The links ends in the wild (sorry, couldn't resist) 404.

**Comment by mickaelistria** [comment]
https://download.eclipse.org/wildwebdeveloper/snapshots/ (without `s` between `sna` and `pshots`)

**Comment by mnlipp** [comment]
Okay. To make absolutely sure, I created a new user account and repeated the steps above. ... and found that I had forgotten to commit the .project files (they are generated files in the full project and therefore were in the .gitignore), sorry.

If you retry with my .project files (added now) you'll notice that `aash-vue-components` is a nested project inside the main project (was mentioned above, but is easy to overlook). This is crucial for reproducing the problem. (If I import only `aash-vue-components`, it work for me as well.) So please check out again and make sure that `aash-vue-components` is imported as nested project.

**Comment by mickaelistria** [comment]
I think it's related to https://github.com/eclipse/wildwebdeveloper/issues/783 which contains steps to reproduce.
